open-turo/action-setup-tools
# [DEVOPS-5720] fix: resolving bugs with java and general improvements/stability/error handling fixes

- refactors ci into common action composites for reuse
- tests both inputs and file based versions
- adds full action test for file based versions
- runs full action tests before releasing
- refactors tools for common logging when checking whether a setup should run
- refactors tools to use `core.exportVariable` instead of modifying process.env directly
- adds env var to control synchronous (easier to debug) or parallel (faster) tool setups
- really really strong error handling in parallel tool setups (probably not doing anything)
- refactor index to allow calling directly from tests without subprocesses
- adds a suite of direct/inline tests for index
- fixes unmarked async tests
- switch naive string parsing of java versions to use regexes instead
- implement check for sdkman having noninteractive settings so ci won't hang
- add generic parser to validateVersion calls
- adds tests for sdkman rc parsing
- adds tests for sdkman config check
- adds tests for validateVersion
- adds `npm run test:debug` for extra output during test runs
- adds tests for individual tools invoked directly with the class to improve coverage
- refactor version() to fix bug and improve handling of all edges
- fix edge case with auto-detecting the tool root directories with command -v
- custom handling of java 1.8.0 version strings to translate to 8.0.x to match sdkman
- adds tests for tool registration
